crud Fabric

tientm03 opened the pull request:
đã hoàn thành:
#19, #20, #21, #24, #25 , #31, #32, #33, #34, #35, #36, #37, #43 , #44, #45

chưa làm:
#22, #23

anhnnhe176642 committed:
Merge pull request #1 from anhnnhe176642/dev

fix role create

anhnnhe176642 committed:
Create Iteration 1 milestone

anhnnhe176642 committed:
Merge branch 'dev'

anhnnhe176642 committed:
Merge branch 'dev'

Commit:
test

Commit:
Merge branch 'dev' of https://github.com/anhnnhe176642/GFWMS_backend into TienTM

Commit:
Làm feature #24 và #25

Commit:
feat(fabric-color): add new fabric color
feat(fabric-color): update existing fabric color
feat(fabric-color): implement fabric color list API

feat(fabric-gloss): add new fabric gloss
feat(fabric-gloss): update existing fabric gloss
feat(fabric-gloss): implement fabric gloss list API

feat(fabric-category): add new fabric category
feat(fabric-category): update existing fabric category
feat(fabric-category): implement fabric category list API

feat(fabric): implement fabric list API
feat(fabric): implement fabric detail API

Closes: #19,#20,#21,#24,#25,#31,#32,#33,#34,#35,#36,#37

Commit:
fix(fabric): resolve incorrect filtering in fabric list API
chore(fabric): remove unnecessary console logs and comments

anhnnhe176642 commented on milestones/Iteration_1.json:
?

anhnnhe176642 commented on src/controllers/fabric.controller.js:
```diff
@@ -0,0 +1,49 @@
+      }
+    });
+
+    console.log('🔍 Query params:', req.query);
```
xoá log đi

tientm03 commented:
Ơ qua xoá rồi push lại lên r mà nhỉ


Vào Th 3, 14 thg 10, 2025 lúc 20:04 Nguyen Ngoc Anh <
***@***.***> đã viết:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In src/controllers/fabric.controller.js
> <https://github.com/anhnnhe176642/GFWMS_backend/pull/41#discussion_r2429088129>
> :
>
> > +import { buildQueryParams } from '../utils/filter-builder.js';
> +
> +/** 🔹 Lấy danh sách Fabric (hỗ trợ filter, sort, pagination) */
> +export const getAllFabrics = async (req, res, next) => {
> +  try {
> +    const queryParams = buildQueryParams(req.query, {
> +      filterFields: ['colorId', 'categoryId', 'glossId', 'supplierId'],
> +      searchFields: ['color.name', 'category.name', 'gloss.name', 'supplier.name'],
> +      dateRangeConfig: {
> +        fromField: 'createdFrom',
> +        toField: 'createdTo',
> +        targetField: 'createdAt'
> +      }
> +    });
> +
> +    console.log('🔍 Query params:', req.query);
>
> xoá log đi
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/anhnnhe176642/GFWMS_backend/pull/41#pullrequestreview-3335505162>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BC6ZUY7OUQID7QH6VDL7IH33XTYF5AVCNFSM6AAAAACJB7GMD6VHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZTGMZVGUYDKMJWGI>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

anhnnhe176642 commented on src/controllers/fabricCategory.controller.js:
```diff
@@ -0,0 +1,77 @@
+  try {
+    const queryParams = buildQueryParams(req.query, {
+      filterFields: [], // nếu muốn filter theo field nào thêm vào đây
+      searchFields: ['name'],
```
tự sáng tạo ra searchFields à

anhnnhe176642 commented on src/controllers/fabricColor.controller.js:
```diff
@@ -0,0 +1,77 @@
+  try {
+    const queryParams = buildQueryParams(req.query, {
+      filterFields: [], 
+      searchFields: ['name'],
```
ko hiểu hàm thì ?

anhnnhe176642 commented on src/controllers/fabricgloss.controller.js:
```diff
@@ -0,0 +1,77 @@
+  try {
+    const queryParams = buildQueryParams(req.query, {
+      filterFields: [], 
+      searchFields: ['description'],
```
thì ctrl + space

anhnnhe176642 commented on src/repositories/fabricCategory.repository.js:
```diff
@@ -0,0 +1,109 @@
+    const where = { AND: [filterWhere] };
+
+    const { skip, take } = buildPagination(page, limit);
+    const orderBy = buildSort(sortBy, order, { name: 'name', createdAt: 'createdAt' });
```
tên input giống nhau thì map làm gì nữa

anhnnhe176642 commented on src/repositories/fabricColor.repository.js:
```diff
@@ -0,0 +1,113 @@
+
+    // Pagination & sort
+    const { skip, take } = buildPagination(page, limit);
+    const orderBy = buildSort(sortBy, order, { name: 'name', createdAt: 'createdAt' });
```
lười đọc hàm thì past cho gpt đọc dùm đc mà

anhnnhe176642 commented on src/repositories/fabricgloss.repository.js:
```diff
@@ -0,0 +1,121 @@
+
+    // Pagination & sort
+    const { skip, take } = buildPagination(page, limit);
+    const orderBy = buildSort(sortBy, order, { description: 'description', createdAt: 'createdAt' });
```
.

anhnnhe176642 replied:
ko hiểu hàm thì ta có thể code chay

anhnnhe176642 replied:
cho hiểu

anhnnhe176642 commented on src/routes/fabric.routes.js:
```diff
@@ -0,0 +1,178 @@
+ *           type: integer
+ *           minimum: 1
+ *           maximum: 100
+ *           default: 10
```
xoá hết validate của swagger đi để test api

anhnnhe176642 commented on src/routes/fabric.routes.js:
```diff
@@ -0,0 +1,178 @@
+router.get(
+  '/',
+  authenticateToken,
+  requirePermission(PERMISSIONS.FABRICS.VIEW_LIST.key),
```
bỏ .key nhìn cho gọn

anhnnhe176642 commented on src/routes/fabric.routes.js:
```diff
@@ -0,0 +1,178 @@
+router.get(
+  '/:id',
+  authenticateToken,
+  requirePermission(PERMISSIONS.FABRICS.VIEW_DETAIL.key),
```
.

anhnnhe176642 commented on src/routes/fabricCategory.routes.js:
```diff
@@ -0,0 +1,269 @@
+ *                 example: Shirt Fabric
+ *               description:
+ *                 type: string
+ *                 maxLength: 255
```
.

anhnnhe176642 commented on src/routes/fabricColor.routes.js:
```diff
@@ -0,0 +1,263 @@
+ *         schema:
+ *           type: integer
+ *           minimum: 1
+ *           default: 1
```
.

Commit:
chore(fabric.routers): remove unnecessary console.log and validation
chore(fabricCategory.routers): remove unnecessary  validation
chore(fabricColor.routers): remove unnecessary  validation
chore(fabricGloss.routers): remove unnecessary  validation

chore(fabric.controller): remove unnecessary  filterFields: []
chore(fabricCategory.controller): remove unnecessary  filterFields: []
chore(fabricColor.controller): remove unnecessary  filterFields: []
chore(fabricGloss.controller):remove unnecessary  filterFields: []

refactor(fabricCategory.repository):rewrite findWithAdvancedQuery function for clarity
refactor(fabricColor.repository): rewrite findWithAdvancedQuery function for clarity
refactor(fabricGloss.repository):rewrite findWithAdvancedQuery function for clarity

Commit:
feat(suppliers): add new supplier
feat(suppliers): update existing supplier
feat(suppliers): implement suppliers list API

Commit:
chore: remove unused file

anhnnhe176642 committed:
Merge branch 'dev' of https://github.com/anhnnhe176642/GFWMS_backend into TienTM

anhnnhe176642 committed:
refactor(fabric.repository): simplify query methods and remove unused functions
